lib/ui/router/app_router_guards.dart

```diff
@@ -1,54 +1,34 @@
-import 'dart:async';
-
-import 'package:auto_route/auto_route.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_template/core/di/di_provider.dart';
 import 'package:flutter_template/core/model/authentication_status.dart';
 import 'package:flutter_template/core/repository/session_repository.dart';
-import 'package:flutter_template/ui/router/app_router.dart';
 
-class _AppAuthStatusGuard extends AutoRouteGuard {
-  final SessionRepository _sessionRepository;
-  final AuthenticationStatus _requiredAppStatus;
-  final PageRouteInfo _redirectPage;
+class AuthenticationScope extends InheritedNotifier<StreamAuthNotifier> {
```

Comment:
cannot we avoid it? I'ts strange to use a InheritedNotifier. We should use a provider or something like that, we shouldn't mix how we get data